public/components/trace_analytics/components/common/plots/service_map.tsx

```diff
@@ -328,33 +291,36 @@ export function ServiceMap({
   };
 
   const onFocus = (service: string) => {
-    if (service.length === 0) {
-      clearFilter();
+    if (!service) {
+      // Clear focus if no service is provided
+      if (focusedService !== null) {
+        removeFilter('serviceName', focusedService);
+        setItems(
+          getServiceMapGraph(
+            serviceMap,
+            idSelected,
+            ticks,
+            undefined,
+            undefined,
+            false // Show the entire graph without filtering
+          )
+        );
+        setFocusedService(null);
+        setInvalid(false);
+      }
     } else if (serviceMap[service]) {
-      // Focus on the specified service and add a filter
-      setFocusedService(service);
-      if (addFilter) {
-        addFilter({
-          field: 'serviceName',
-          operator: 'is',
-          value: service,
-          inverted: false,
-          disabled: false,
-        });
+      if (focusedService !== service) {
+        const filteredGraph = getServiceMapGraph(
+          serviceMap,
+          idSelected,
+          ticks,
+          service,
+          serviceMap[service]?.relatedServices,
+          true // Enable filtering to focus on connected nodes
```

Comment:
i didn't look into `getServiceMapGraph`, but seems `service` and `serviceMap[service]?.relatedServices` are used for filtering, and they will be absent if we don't want filtering. Then is this boolean required?

Reply:
The focus functionality is not applying a filter so for the new experience these are required to change the options that show up using the focus functionality.

Without:

https://github.com/user-attachments/assets/d560e8d1-83c6-4c98-ad40-6b1024dce7f5

With: 

https://github.com/user-attachments/assets/b5fa8fef-eab6-416d-bf5c-b53cfc80607c